sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/CoroutineScopeFunSuspendingCheck.kt

```diff
@@ -0,0 +1,36 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.js.descriptorUtils.getJetTypeFqName
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.psiUtil.isExtensionDeclaration
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.typeUtil.supertypes
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.api.SecondaryLocation
+import org.sonarsource.kotlin.api.suspendModifier
+import org.sonarsource.kotlin.converter.KotlinTextRanges.textRange
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val COROUTINE_SCOPE = "kotlinx.coroutines.CoroutineScope"
+private const val MESSAGE = "Extension functions on CoroutineScope should not be suspending"
+
+@Rule(key = "S6312")
+class CoroutineScopeFunSuspendingCheck : AbstractCheck() {
+    override fun visitNamedFunction(function: KtNamedFunction, kotlinFileContext: KotlinFileContext) {
+        // Only applicable for extension functions
+        if (!function.isExtensionDeclaration()) return
+
+        val suspendModifier = function.suspendModifier() ?: return
+        val receiverType = function.receiverTypeReference ?: return
```

Comment:
It looks like this condition cannot fail, because of line L22. You can simplify it a little bit and remove the check on L22 at all